ndmg/stats/qa_skullstrip.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python
+
+# Copyright 2016 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+# Created by Vikram Chandrashekhar.
+# Edited by Greg Kiar, Eric Bridgeford and Chuankai Luo.
+# Email: Greg Kiar @ gkiar@jhu.edu   cluo16@jhu.edu
+
+import warnings
+
+warnings.simplefilter("ignore")
+import os
+import re
+import sys
+import numpy as np
+import nibabel as nb
+# import ndmg.utils as mgu
+from argparse import ArgumentParser
+from scipy import ndimage
+from matplotlib.colors import LinearSegmentedColormap
+import matplotlib as mpl
+
+mpl.use("Agg")  # very important above pyplot import
+# from nilearn.plotting.edge_detect import _edge_map as edge_map
+import matplotlib.pyplot as plt
+from ndmg.utils.qa_utils import get_min_max, opaque_colorscale,pad_im
+from ndmg.utils.gen_utils import get_filename, get_braindata
+
+
+
+
+
+def gen_overlay_pngs(
+    brain, original, outdir, loc=0, mean=False, minthr=2, maxthr=95, edge=False):
+    """
+    generate the image combine the brain and original data
+
+    Parameters
+    ----------
+    brain:
+        is the skull striped result, only left the brain
+    original:
+        is the original nii.gz file
+    outdir:
+        directory where output png file is saved
+    loc and mean is to deal with 4d data
+    minthr, maxthr, edge are the variables about the color
```

Comment:
this docstring / function name needs to be fixed.
- function name should be more descriptive than `gen_overlay_pngs` if it's for skullstrip specifically. Make the name have the word "skullstrip" in it.
- add types in front of parameters (look at other docstrings the pipeline to see what I mean -- e.g., `brain : str`
- description should be updated (the english wording is a little confusing 🙂). Something like "Generate a QA image for skullstrip. Shows the skullstripped brain (green) overlayed on the original t1w (pink)".
- descriptions for each parameters also need to be updated. e.g., for `brain`, `"Path to the skull-stripped nifti brain."`, for `original`, `"Path to the original t1w brain, with the skull included."`, and for `outdir`, `"Path to the directory where QA will be saved."`.
- `loc`, `mean`, `minthr`, `maxthr`, and `edge` need their own parameter places (like `brain`, `original`, etc)

Reply:
update: just saw `plot_overlays_skullstrip`. Have docstring say that this function is a wrapper on that one.

Reply:
@luochuankai-JHU We use [numpy-formatted docstrings](https://numpydoc.readthedocs.io/en/latest/format.html), so please use that format on everything.

I have a vscode extension called [autoDocstring](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) that I find helpful for this. You can make it use numpy formatting in the extension settings.

Reply:
fixed it

Reply:
Thanks a lot